File: Java/libraries/recognizers-text/src/main/java/com/microsoft/recognizers/text/matcher/AaNode.java
package com.microsoft.recognizers.text.matcher;

import java.util.HashMap;
import java.util.Iterator;

public class AaNode<T> extends Node<T> {
    T word;
    int depth;
    AaNode<T> parent;
    AaNode<T> fail;

    public AaNode(T word, int depth, AaNode<T> parent) {
        this.word = word;
        this.depth = depth;
        this.parent = parent;
        this.fail = null;
    }

    public AaNode(T word, int depth) {
        this(word, depth, null);
    }

    public AaNode() {
        this(null, 0);
    }

    AaNode<T> get(T c) {
        return children != null && children.containsKey(c) ? (AaNode<T>)children.get(c) : null;
    }

    void put(T c, AaNode<T> value) {
        if (children == null) {
            children = new HashMap<>();
        }

        children.put(c, value);
    }

    @Override
    Iterator getEnumerator() {
        return children != null ? children.values().stream().map(x -> (AaNode<T>)x).iterator() : null;
    }

    @Override
    Iterable getIterable() {
        return  children != null ? (Iterable)children.values().stream().map(x -> (AaNode<T>)x) : null;
    }

    @Override
    public String toString() {
        return word.toString();
    }
}

File: Python/libraries/recognizers-text/recognizers_text/Matcher/AaNode.py
from .Node import Node


class AaNode(Node):

    def __init__(self, c: [] = [], depth: int = 0, parent=[]):
        self.__word = c,
        self.__depth = depth,
        self.__parent = parent,
        self.__fail = 0

    @property
    def word(self) -> []:
        return self.__word

    @word.setter
    def word(self, word) -> []:
        self.__word = word

    @property
    def depth(self) -> int:
        return self.__depth

    @depth.setter
    def depth(self, depth) -> int:
        self.__depth = depth

    @property
    def parent(self):
        return self.__parent

    @parent.setter
    def parent(self, parent):
        self.__parent = parent

    @property
    def fail(self):
        return self.__fail

    @fail.setter
    def fail(self, fail):
        self.__fail = fail

    def get_enumerator(self):
        pass

    def to_string(self):
        pass
